universal-application-tool-0.0.1/app/views/applicant/ApplicantInformationView.java

```diff
@@ -0,0 +1,77 @@
+package views.applicant;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.form;
+import static j2html.TagCreator.p;
+
+import com.google.common.collect.ImmutableList;
+import controllers.applicant.routes;
+import j2html.tags.ContainerTag;
+import java.util.AbstractMap;
+import java.util.Locale;
+import javax.inject.Inject;
+import play.i18n.Lang;
+import play.i18n.Langs;
+import play.mvc.Http;
+import play.twirl.api.Content;
+import views.BaseHtmlView;
+import views.components.SelectWithLabel;
+
+/**
+ * Provides a form for selecting an applicant's preferred language. Note that we cannot use Play's
+ * {@link play.i18n.Messages}, since the applicant has no language set yet. Instead, we use English
+ * since this is the CiviForm default language.
+ */
+public class ApplicantInformationView extends BaseHtmlView {
+
+  private final ApplicantLayout layout;
+  private final ImmutableList<Locale> supportedLanguages;
+
+  @Inject
+  public ApplicantInformationView(ApplicantLayout layout, Langs langs) {
+    this.layout = layout;
+    this.supportedLanguages =
+        langs.availables().stream().map(Lang::toLocale).collect(toImmutableList());
+  }
+
+  public Content render(Http.Request request, long applicantId) {
+    String formAction = routes.ApplicantInformationController.update(applicantId).url();
+    return layout.render(
+        form()
+            .withAction(formAction)
+            .withMethod(Http.HttpVerbs.POST)
+            .with(makeCsrfTokenInputTag(request))
+            .with(selectLanguageDropdown())
+            .with(submitButton("Submit")));
+  }
+
+  private ContainerTag selectLanguageDropdown() {
+    SelectWithLabel languageSelect = new SelectWithLabel();
```

Comment:
suggestion (non-blocking): I'm not sure this is necessarily preferable, but I believe this could be written like:

```suggestion
    SelectWithLabel languageSelect = new SelectWithLabel()
            .setId("select-language")
            .setFieldName("locale")
            .setOptions(...);
```

Reply:
Great suggestion! Fixed